PowerPointLabs/PowerPointLabs/CropLab/CropToSlide.cs

```diff
@@ -0,0 +1,229 @@
+                maxX = Math.Max(rotated.X, maxX);
+                maxY = Math.Max(rotated.Y, maxY);
+            }
+            return new RectangleF(shape.Left + shape.Width / 2 + minX, shape.Top + shape.Height / 2 + minY,
+                                  maxX - minX, maxY - minY);
+        }
+
+        private static PointF RotatePoint(PointF point, float theta)
+        {
+            return new PointF((float)(point.X * Math.Cos(theta) - point.Y * Math.Sin(theta)),
+                            (float)(point.X * Math.Sin(theta) + point.Y * Math.Cos(theta)));
+        }
+        private static RectangleF GetCropArea(PowerPoint.Shape shape, float slideWidth, float slideHeight)
+        {
+            float cropTop = Math.Max(0, shape.Top);
+            float cropLeft = Math.Max(0, shape.Left);
+            float cropHeight = shape.Height - Math.Max(0, -shape.Top);
+            float cropWidth = shape.Width - Math.Max(0, -shape.Left);
+
+            cropHeight = Math.Min(slideHeight - cropTop, cropHeight);
+            cropWidth = Math.Min(slideWidth - cropLeft, cropWidth);
+
+            return new RectangleF(cropLeft, cropTop, cropWidth, cropHeight);
+        }
+
+        private static bool VerifyIsShapeRangeValid(PowerPoint.ShapeRange shapeRange, bool handleError)
+        {
+            try
+            {
+                if (shapeRange.Count < 1)
+                {
+                    ThrowErrorCode(ErrorCodeForSelectionCountZero);
+                }
+
+                if (!IsPictureForSelection(shapeRange))
+                {
+                    ThrowErrorCode(ErrorCodeForSelectionNonPicture);
+                }
+
+                return true;
+            }
+            catch (Exception e)
+            {
+                if (handleError)
+                {
+                    ProcessErrorMessage(e);
+                    return false;
+                }
+
+                throw;
+            }
+        }
+
+        private static void VerifyIsSelectionValid(PowerPoint.Selection selection)
+        {
+            if (selection.Type != PowerPoint.PpSelectionType.ppSelectionShapes)
+            {
+                ThrowErrorCode(ErrorCodeForSelectionCountZero);
+            }
+        }
+
+        private static bool IsPictureForSelection(PowerPoint.ShapeRange shapeRange)
+        {
+            return (from PowerPoint.Shape shape in shapeRange select shape).All(IsPicture);
+        }
+
+        private static bool IsPicture(PowerPoint.Shape shape)
+        {
+            return shape.Type == Office.MsoShapeType.msoPicture ||
+                   shape.Type == Office.MsoShapeType.msoLinkedPicture;
+        }
+
+        private static void ThrowErrorCode(int typeOfError)
+        {
+            throw new Exception(typeOfError.ToString(CultureInfo.InvariantCulture));
+        }
+
+        private static void IgnoreExceptionThrown() { }
+
+        public static string GetErrorMessageForErrorCode(string errorCode)
```

Comment:
dont repeat yourself, saw it many times.
have you consider making a parent class for them